src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Commands/EditApprenticeship/EditApprenticeshipCommandHandler.cs

```diff
@@ -0,0 +1,143 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Domain.Exceptions;
+using SFA.DAS.CommitmentsV2.Domain.Interfaces;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.CommitmentsV2.Data.Extensions;
+using SFA.DAS.CommitmentsV2.Models;
+using SFA.DAS.CommitmentsV2.Authentication;
+using SFA.DAS.CommitmentsV2.Types;
+using SFA.DAS.CommitmentsV2.Domain.Extensions;
+using SFA.DAS.CommitmentsV2.Application.Queries.GetTrainingProgramme;
+
+namespace SFA.DAS.CommitmentsV2.Application.Commands.EditApprenticeship
+{
+    public class EditApprenticeshipCommandHandler : IRequestHandler<EditApprenticeshipCommand, EditApprenticeshipResponse>
+    {
+        private readonly Lazy<ProviderCommitmentsDbContext> _dbContext;
+        private readonly IEditApprenticeshipValidationService _editValidationService;
+        private readonly IAuthenticationService _authenticationService;
+        private readonly ICurrentDateTime _currentDateTime;
+        private readonly ILogger<EditApprenticeshipCommandHandler> _logger;
+        private readonly IMediator _mediator;
+
+        public EditApprenticeshipCommandHandler(IEditApprenticeshipValidationService editValidationService, Lazy<ProviderCommitmentsDbContext> dbContext, IAuthenticationService authenticationService, ICurrentDateTime currentDateTime,  IMediator mediator, ILogger<EditApprenticeshipCommandHandler> logger)
+        {
+            _editValidationService = editValidationService;
+            _dbContext = dbContext;
+            _authenticationService = authenticationService;
+            _currentDateTime = currentDateTime;
+            _logger = logger;
+            _mediator = mediator;
+        }
+
+        public async Task<EditApprenticeshipResponse> Handle(EditApprenticeshipCommand command, CancellationToken cancellationToken)
+        {
+            var party = _authenticationService.GetUserParty();
+            var apprenticeship = await _dbContext.Value.GetApprenticeshipAggregate(command.EditApprenticeshipRequest.ApprenticeshipId, cancellationToken);
+
+            await Validate(command, apprenticeship, party, cancellationToken);
+
+            CreateImmedidateUpdate(command, party, apprenticeship);
```

Comment:
returned value here is never assigned or used? Do we need to create the intermediate update if there has been an immediate one?